doc/how-to/how-to-setup-custom-subdomains.md

```diff
@@ -0,0 +1,87 @@
+
+    The `<TEAM_NAME>.key` file should be kept, we require this for step XXX.
+
+
+3. **IT Team** - Provide PlanX with certificates (details in documentation above)
+    - Certificates can be provided as individual files, or in an archive format (`.pfx`, `.pkcs12`, `.p12`)
+
+4. **PlanX** - Format certificates if provided with PKCS #12
+    ```shell
+    openssl pkcs12 -nocerts -nodes -in <FILENAME> -out council.key
+    openssl pkcs12 -nokeys -in <FILENAME> -out council.cert
+    ```
+    
+    The `.cert` file might contain the certificate chain inside it, so please separate the first certificate in the file (the certificate body) from the rest of the certificates in the file (the certificate chain).
+
+5. **PlanX** - Validate certificates manually in AWS Console (optional but recommended)
+  
+    To avoid slow round trip loops of Pizza -> Staging -> Production, certificates can be manually imported via the AWS Console to check their validity before proceeding.
+
+    In order to do this -
+
+      - Log into AWS staging 
+      - Navigate to AWS Certificate Manager (ACM)
+      - Change to region to us-east-1 (N.Virginia) (required for CloudFront)
+      - Import certificate (copy cert, key and chain into form)
+      - Check the following after import - 
+        - Subdomain is correct ✅
+        - Key algorithm is RSA 2048 ✅
+        - Navigate to AWS CloudFront > Create new distribution. The certificate is listed in the dropdown for "Custom SSL certificate" ✅
+      - If the above checks pass, proceed to next step
+
+6. **PlanX** - Add certificates to AWS infrastructure
+
+    Certificates and keys are added to our infrastructure as Pulumi secrets which are then read to generate an ACM record and CloudFront distribution for the custom domain when a deployment is made to Production.
+
+    1. Add the team to the `CUSTOM_DOMAINS` array in `infrastructure/application/index.ts`
+
+    2. Add secrets to Pulumi
+
+        The following secrets need to be added to the Pulumi application layer - `ssl-{team}-key`, `ssl-{team}-cert` and `ssl-{team}-chain` (optional). These secrets are only required for the production stack.
+
+        The provided files, or output of step 4, can be piped directly into these variables.
+
+        ```bash
+        cd infrastructure/application
+
+        cat council.key | pulumi config set ssl-{team}-key --stack production --secret
+
+        cat council.cert | pulumi config set ssl-{team}-cert --stack production --secret
+
+        cat chain.cert | pulumi config set ssl-{team}-chain --stack production --secret
+        ```
+
+        To validate the above steps, check `infrastructure/application/Pulumi.production.yaml` which will list the above secrets for the team in an encrypted format.
+
+        At this stage we can execute `pulumi preview` from the application layer to check for any issues with the config.
```

Comment:
is it true that CI will execute this when you open a PR? or were you running this command manually? (or both: eg doing manually is quicker feedback loop for any issues than CI, but CI also always runs this to confirm?)

difference of which CI jobs are running `pulumi preview` versus `pulumi up` is usually one I need to double check so being explicit in docs is helpful for me!

Reply:
Thanks for raising this one. It wasn't actually part of my process, but something I actually found in Gunar's docs when pulling this together and thought it might be helpful. I kicked myself when I saw it as I thought it could have saved some time / frustration.

I checked this out and it turns out it isn't really very useful here so I've dropped it from the docs for now.

`pulumi preview` can be run locally and does run on all PRs but it just gives a diff of what will change. Here's [a live preview](https://app.pulumi.com/planx/application/staging/previews/dda561c8-f96f-4f7b-829f-1048c2d2bb5e) from a PR which later failed `pulumi up` when deploying to prod. On the PR into `main` it's running `pulumi preview --stack staging` which isn't helpful when making a change to production.

Running `pulumi preview --stack production` is failing for me locally on both the application and data layer.

Here's what I propose - 
 - Work out and fix `pulumi preview --stack production` locally
 - Test locally with invalid certs and see if this is helpful. If so, add back to docs.
 - Run this as part of CI when opening a PR into `production` - there are parts of our IaC which treat production and staging stacks differently so this is something we would want to be doing

Reply:
totally makes sense, thanks for clearly outlining here - I confirmed same local behavior on `pulumi preview --stack production` :heavy_plus_sign: